src/disasm_wrapper.c

```diff
@@ -0,0 +1,291 @@
+	 * to parse the resulting string.
+	 */
+	if (cs_option(context->handle, CS_OPT_DETAIL, CS_OPT_ON) != 0)
+		xabort();
+
+	/*
+	 * Overriding the printing routine used by capstone,
+	 * see comments above about nop_vsnprintf.
+	 */
+	cs_opt_mem x = {
+		.malloc = malloc,
+		.free = free,
+		.calloc = calloc,
+		.realloc = realloc,
+		.vsnprintf = nop_vsnprintf};
+	if (cs_option(context->handle, CS_OPT_MEM, (size_t)&x) != 0)
+		xabort();
+
+	if ((context->insn = cs_malloc(context->handle)) == NULL)
+		xabort();
+
+	return context;
+}
+
+/*
+ * intercept_disasm_destroy -- see comments for above routine
+ */
+void
+intercept_disasm_destroy(struct intercept_disasm_context *context)
+{
+	cs_free(context->insn, 1);
+	cs_close(&context->handle);
+	xmunmap(context, sizeof(*context));
+}
+
+/*
+ * check_op - checks a single operand of an instruction, looking
+ * for RIP relative addressing.
+ */
+static void
+check_op(struct intercept_disasm_result *result, cs_x86_op *op,
+		const unsigned char *code)
+{
+	/*
+	 * the addres the RIP register is going to contain during the
+	 * execution of this instruction
+	 */
+	const unsigned char *rip = code + result->length;
+
+	if (op->type == X86_OP_REG) {
+		if (op->reg == X86_REG_IP ||
+				op->reg == X86_REG_RIP) {
+			/*
+			 * Example: mov %rip, %rax
+			 */
+			result->has_ip_relative_opr = true;
+			result->rip_disp = 0;
+			result->rip_ref_addr = rip;
+		}
+		if (result->is_jump) {
+			/*
+			 * Example: jmp *(%rax)
+			 */
+			/*
+			 * An indirect jump can't have arguments other
+			 * than a register - therefore the asserts.
+			 * ( I'm 99.99% sure this is true )
+			 */
+			assert(!result->is_rel_jump);
+			result->is_indirect_jump = true;
+		}
+	} else if (op->type == X86_OP_MEM) {
+		if (op->mem.base == X86_REG_IP ||
+				op->mem.base == X86_REG_RIP ||
+				op->mem.index == X86_REG_IP ||
+				op->mem.index == X86_REG_RIP ||
+				result->is_jump) {
+			result->has_ip_relative_opr = true;
+			assert(!result->is_indirect_jump);
+
+			if (result->is_jump)
+				result->is_rel_jump = true;
+
+			assert(op->mem.disp <= INT32_MAX);
+			assert(op->mem.disp >= INT32_MIN);
+
+			result->rip_disp = (int32_t)op->mem.disp;
+			result->rip_ref_addr = rip + result->rip_disp;
+		}
+	} else if (op->type == X86_OP_IMM) {
+		if (result->is_jump) {
+			assert(!result->is_indirect_jump);
+			result->has_ip_relative_opr = true;
+			result->is_rel_jump = true;
+			result->rip_ref_addr = (void *)op->imm;
+
+			result->rip_disp =
+			    (int32_t)((unsigned char *)op->imm - rip);
+		}
+	}
+}
+
+/*
+ * intercept_disasm_next_instruction - Examines a single instruction
+ * in a text section. This is only a wrapper around capstone specific code,
+ * collecting data that can be used later to make decisions about patching.
+ */
+struct intercept_disasm_result
+intercept_disasm_next_instruction(struct intercept_disasm_context *context,
+					const unsigned char *code)
+{
+	(void) context;
```

Comment:
Is this necessary? The intent of this, as I understand it, is to shut off compiler warnings about an unused parameter. But, you are using the 'context' parameter.